round4/Team-4.md

```diff
@@ -0,0 +1,72 @@
+7. **Performant**: Our web blocker is optimized to run smoothly without slowing down the user's browser.
+
+## User Stories
+
+1. As a student, I want to block social media websites during my study sessions to maintain focus and improve my academic performance.
+2. As a parent, I want to set up time limits and block inappropriate content on my child's devices to ensure a safe and productive online experience.
+3. As a remote worker, I want to block non-work-related websites during designated work hours to create a more disciplined and productive routine.
+4. As a gamer, I want to block gaming forums and websites during my work periods to minimize distractions and improve my academic performance.
+5. As a creative professional, I want to block self-critical or negative websites during my creative sessions to foster a more positive and productive mindset.
+6. As a digital minimalist, I want to block unnecessary websites and limit my online time to reduce digital clutter and increase mindfulness.
+7. As a meditation practitioner, I want to block news websites and social media platforms during my meditation sessions to create a serene and distraction-free environment.
+8. As a cybersecurity-conscious individual, I want to block suspicious or phishing websites to protect my personal information and enhance online security.
+9. As a recovering addict, I want to block websites related to addictive substances or behaviors to support my journey towards sobriety and wellness.
+10. As a financial planner, I want to block impulse-buying websites during my budgeting sessions to stick to my financial goals and avoid unnecessary expenses.
+11. As a researcher, I want to block social media and news websites during my research sessions to stay focused and avoid information overload.
+12. As a productivity coach, I want to recommend personalized website blocking settings to my clients based on their specific goals and challenges.
+13. As an individual looking to improve my mental health, I want to block triggering or distressing websites during vulnerable moments to protect my emotional well-being and promote self-care.
+14. As a developer, I require a feature in our productivity tool to block access to non-relevant websites during work hours, ensuring my focus remains on programming tasks and problem-solving without distractions.
+15. As a founder, I need a web blocking extension to help me stay focused on strategic tasks and minimize distractions during critical business operations.
+
+## Notes
+
+### Modes
+1. **À La Carte**: Set and name a time period and a set of block lists to prevent access to these websites during this time.
+2. **Scheduled**: Set and name a recurring time period and a set of block lists to prevent access during this time.
+3. **Work**: Alternate between periods of named blocking (work) and unblocking (breaks) with times set dynamically and with a persistent but modifiable set of block lists.
+
+### Block Lists
```

Comment:
I think this concept of block list is useful and central to your extension and value proposition.
- It is a good idea, to have some "preconfigured block lists" that a user — a testing user, or a busy user — could use off the shelf.
- It is also useful to think about the range of patterns: Are users entering domain names? keywords to be matched? regular expressions? Maybe looking at how Ad Block and other products that maintain block list do it, might be instructive?
- The interface to enter website/patterns is going to be one of the most important features probably. A lot of interfaces are confusing, and the source of misunderstanding with the user (i.e., the user doesn't understand what they are doing, and misuse the extension, and think the extension is broken — incorrectly so).

Reply:
I think we'd like to start with a more conservative blocking approach (domain-level) then move onto a less restrictive UI for DNS navigation in URLs that gives users the option to block specific pages of a site (youtube.com/shorts, for example). We'd ideally like to hide as much of this URL sort of thing as possible because it's not aesthetic and not always useful, and leave it as an advanced setting that can be accessed in the detailed view of block lists.